### PR TITLE
RFC 53: updates to RFC process

### PR DESCRIPTION
**Pull request by eladb**
As a follow up to a retrospective the CDK team has performed about the RFC process. 

This PR proposes the following changes:

* Decouple the creation of the tracking issue from the submission of the RFC pull request to simplify the initial phase.
* Add a concept of "API Bar Raiser" who needs to approve the "working backwards" section of the RFC
* Remove RFC PR # from document - the tracking issue is the RFC number.
* Simplify template a little
* Add details in the main README which describes each step
* Additional minor changes

See https://github.com/cdklabs/team-internal/pull/18 for internal process changes.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/tracking-issue.md
Comment (eladb) [inline .github/ISSUE_TEMPLATE/tracking-issue.md:44]:
```suggestion
> Driver is responsible to progress the RFC according to this checklist, and apply the
```

File: README.md
Comment (nija-at) [inline README.md:179]: As mentioned above, we should describe how to find people for these roles.

Moreover, the first part of this README says that 'anyone' can create an RFC. If so, we need a lot more guidance for contributors not on the team.

.... Or, we reduce the scope and say that RFCs are created by the CDK team. (we would still need some guidance in this paragraph)

File: README.md
Comment (nija-at) [inline README.md:198]: Similar comment as above.

File: README.md
Comment (nija-at) [inline README.md:200]:
```suggestion
Before diving into writing the RFC, it is highly recommended to organize a
```

File: README.md
Comment (nija-at) [inline README.md:205]: My suggestion would be to do something different. It would be to identify 'some' high level approaches, put that down in the RFC or tracking issue, and then have that kickoff meeting. 

This way, there is some starting point and people don't arrive at a meeting with a blank slate.

File: README.md
Comment (nija-at) [inline README.md:296]: This section needs updating as well.

**Comment by eladb** [comment]
From our conversation amongst CDK tech leads: we will reframe the RFC process to use an “API Bar Raiser” model instead of an approver model.  Every RFC will get assigned a “tier” and bar raiser for API Bar Raising by the CDK Tech Leadership (email aws-cdk-tech-leads@).

* T1 - Launches (L7)
* T2 - Core AWS CDK APIs, incl. JSII CLIs (L6+)
* T3 - AWS Construct Library APIs (L5+)

The alias of the bar raiser will be included in the RFC document itself and the same bar raiser will be required if major API changes are performed.

As for the rest of the RFC, the RFC PR will undergo the normal PR process of the CDK team, which means that an approver from the core team will eventually need to approve the PR in order to merge it. The API bar raiser will need to sign-off on the “working backwards” section, which primarily represents the public API of the feature.

File: .github/ISSUE_TEMPLATE/tracking-issue.md
Comment (eladb) [reply .github/ISSUE_TEMPLATE/tracking-issue.md:12]: Simplified

File: .github/ISSUE_TEMPLATE/tracking-issue.md
Comment (eladb) [reply .github/ISSUE_TEMPLATE/tracking-issue.md:24]: Added a link to the RFC Process documentation

File: .github/ISSUE_TEMPLATE/tracking-issue.md
Comment (eladb) [reply .github/ISSUE_TEMPLATE/tracking-issue.md:35]: Not sure how to phrase this here. Maybe we can add some info about periods in the README. Thoughts?

File: README.md
Comment (eladb) [reply README.md:179]: Modified to reframe as "API Bar Raiser" - let me know what do you think

File: README.md
Comment (eladb) [reply README.md:296]: Not sure - it's generally the same state flow

File: README.md
Comment (eladb) [reply README.md]: I'll add a recommendation to write something up in the tracking issue as a prep for that meeting

File: README.md
Comment (BenChaimberg) [inline README.md:203]: I'm confused about whether the bar raiser is assigned by the team or the author should reach out to get one assigned

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (nija-at) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: nit: I found this link didn't quite work last time I tried. The relative url was broken I think, but I can't recall exactly.

File: 0000-template.md
Comment (nija-at) [inline 0000-template.md:1]: minor: Not a fan of this format. I prefer a separate line above or below this heaeder linking to the tracking issue.

File: README.md
Comment (nija-at) [inline README.md:193]: The definition core team is changing quite a bit with the two teams, etc. Should we say 'cdk leads' instead?

File: README.md
Comment (nija-at) [inline README.md:195]:
```suggestion
exposed to the feature. As a general rule, for rfcs with "significant" changes,
```

File: README.md
Comment (nija-at) [inline README.md:204]: We have not established a regular practice of monitoring this. It shouldn't rely on something one person does.

We should include this as part of our regular 'triage' process, included into our usual monitoring to ensure incoming issues get triaged.

File: .github/ISSUE_TEMPLATE/tracking-issue.md
Comment (nija-at) [inline .github/ISSUE_TEMPLATE/tracking-issue.md:30]: >> How long should these periods be? Worth adding here as a reminder.
>> 
>> One of the items we discussed was making the RFC process quick. I would personally prefer for simple ones to not exceed 5 days or a week.
>
> Not sure how to phrase this here. Maybe we can add some info about periods in the README. Thoughts?

Could be a good start yeah. Each state in the README which needs active waiting (like 'final comments period', 'waiting for feedback', etc.) can have a clear subtext or highlight of on what's the required max waiting period. Some (most?) rfcs may not receive any community input.

Then link each section in the tracking issue to the anchor link in the README?

File: README.md
Comment (nija-at) [inline README.md:223]: I generally prefer doing this in the RFC rather than tracking issue. Basically a rough half baked (or not baked) rfc. 

It allows me to iterate on my idea via git, before I'm ready to share.

Can we say either or move to that model? I generally find doing this on issues to be a little hard to manage.

File: README.md
Comment (nija-at) [inline README.md:290]: why not simply add a label?

File: README.md
Comment (nija-at) [inline README.md:194]: Where do you plan to document the tiering model and how they're determined?

File: README.md
Comment (eladb) [inline README.md:135]: add: The goal of the RFC is to reduce risk and ambiguity. One question that you should be able to answer after an RFC is ready is how long will it take to implement this in production quality. Do you both feel you flushed out enough of the design to be able to provide a confident answer?

File: .github/ISSUE_TEMPLATE/tracking-issue.md
Comment (eladb) [reply .github/ISSUE_TEMPLATE/tracking-issue.md:30]: Final comments period already indicates a `~week`. I've added a small note about feedback. I think that's good enough for now.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (eladb) [reply .github/PULL_REQUEST_TEMPLATE.md:4]: Removed it

File: 0000-template.md
Comment (eladb) [reply 0000-template.md:1]: Fixed

File: README.md
Comment (eladb) [reply README.md:135]: Added

File: README.md
Comment (eladb) [reply README.md:193]: Fixed

File: README.md
Comment (eladb) [reply README.md:194]: Our internal team flow documents

File: README.md
Comment (eladb) [reply README.md:195]: rephrased a bit

File: README.md
Comment (eladb) [reply README.md:203]: Either option is good. This is targeting both internal and external folks, so I wanted to give both options.

File: README.md
Comment (eladb) [reply README.md:204]: I'll add to team flow

File: README.md
Comment (eladb) [reply README.md:223]: I think some initial iteration on the issue is desirable (and actually came up in our retrospective as a light-weight process to get started).

Comment (eladb) [reply README.md:223]: Changed to "you can..."

File: README.md
Comment (eladb) [reply README.md:290]: Good call

Comment (eladb) [reply README.md:290]: Fixed.

File: README.md
Comment (eladb) [reply README.md]: https://github.com/cdklabs/team-internal/pull/18

File: .github/ISSUE_TEMPLATE/tracking-issue.md
Comment (RomainMuller) [inline .github/ISSUE_TEMPLATE/tracking-issue.md:17]: Strongly suggest opening up the process to **collaborative** editing. It feels lonely right now.

```suggestion
| Author(s)           | @alias, @alias, @alias
```

File: 0000-template.md
Comment (RomainMuller) [inline 0000-template.md:3]:
```suggestion
* **Original Author(s):**: @{AUTHOR}
```

File: README.md
Comment (RomainMuller) [inline README.md:220]: Perhaps suggest how to go about this? From an external collaborator's perspective, it is not clear how to get a hold of people's calendars, what conferencing solutions are available - which are free to use? which are acceptable?

Comment (eladb) [reply README.md:220]: Good call. I'll add a suggestion to reach out to the team via the #aws-cdk-rfcs slack channel to get help organizing this

Comment (eladb) [reply README.md:220]: Added

**Review by skinny85 — COMMENTED** [review]
Looks good! I think the new ["StackSets support" RFC](https://github.com/aws/aws-cdk-rfcs/pull/347) might be a good candidate to try out this new process on.

I guess the only irritating part is that the template is changing, so the author might have to do some work to move to the new template...

**Comment by eladb** [comment]
> Looks good! I think the new "StackSets support" RFC might be a good candidate to try out this new process on.

Yes, let's apply this scheme to this RFC!

**Review by RomainMuller — APPROVED** [review]

**Pull request merged**
